store/src/lib.rs

```diff
@@ -44,20 +44,22 @@ use byteorder::{BigEndian, WriteBytesExt};
 pub use crate::lmdb::*;
 
 /// Build a db key from a prefix and a byte vector identifier.
-pub fn to_key(prefix: u8, k: &mut Vec<u8>) -> Vec<u8> {
+pub fn to_key<K: AsRef<[u8]>>(prefix: u8, k: K) -> Vec<u8> {
+	let k = k.as_ref();
 	let mut res = Vec::with_capacity(k.len() + 2);
 	res.push(prefix);
 	res.push(SEP);
-	res.append(k);
+	res.extend_from_slice(k);
```

Comment:
Okay, interesting. So instead of emptying `k` we clone and append the slice. So we don't need a mutable reference 👍

Reply:
I think in general it's preferable to use `extend` over `append`, because the later consumes the param. `extend_from_slice` is just an optimized version which may be removed when Rust gets specialization.